x-pack/plugins/endpoint/public/applications/endpoint/store/endpoint_list/saga.ts

```diff
@@ -7,16 +7,28 @@
 import { CoreStart } from 'kibana/public';
 import { SagaContext } from '../../lib';
 import { EndpointListAction } from './action';
+import { endpointListPageIndex, endpointListPageSize } from './selectors';
 
 export const endpointListSaga = async (
   { actionsAndState, dispatch }: SagaContext<EndpointListAction>,
   coreStart: CoreStart
 ) => {
   const { post: httpPost } = coreStart.http;
 
-  for await (const { action } of actionsAndState()) {
-    if (action.type === 'userEnteredEndpointListPage') {
-      const response = await httpPost('/api/endpoint/endpoints');
+  for await (const { action, state } of actionsAndState()) {
```

Comment:
@paul-tavares @kevinlog @parkiino Are we going to go with the `oatSaga` approach or with the simple `middleware` approach? We discussed using the simpler approach until we really needed to add complexity. While the alert and endpoint pages can go in different directions, I personally would favor consistency.

Reply:
no, we'll refactor to use Redux middleware. this is also going to be done under elastic/endpoint-app-team/issues/131